sklearn/decomposition/pca.py

```diff
@@ -96,49 +102,78 @@ def _infer_dimension_(spectrum, n_samples, n_features):
 
-    The time complexity of this implementation is ``O(n ** 3)`` assuming
-    n ~ n_samples ~ n_features.
+    It uses the scipy.linalg ARPACK implementation of the SVD or a randomized
+    SVD by the method of Halko et al. 2009, depending on which is the most
+    efficient.
 
     Read more in the :ref:`User Guide <PCA>`.
 
     Parameters
     ----------
-    n_components : int, None or string
+    n_components : int, float, None or string
         Number of components to keep.
         if n_components is not set all components are kept::
 
             n_components == min(n_samples, n_features)
 
-        if n_components == 'mle', Minka\'s MLE is used to guess the dimension
-        if ``0 < n_components < 1``, select the number of components such that
-        the amount of variance that needs to be explained is greater than the
-        percentage specified by n_components
+        if n_components == 'mle' and svd_solver == 'full', Minka\'s MLE is used
+        to guess the dimension
+        if ``0 < n_components < 1`` and svd_solver == 'full', select the number
+        of components such that the amount of variance that needs to be
+        explained is greater than the percentage specified by n_components
+        n_components cannot be equal to n_features for svd_solver == 'arpack'.
 
-    copy : bool
+    copy : bool (default True)
         If False, data passed to fit are overwritten and running
         fit(X).transform(X) will not yield the expected results,
         use fit_transform(X) instead.
 
-    whiten : bool, optional
-        When True (False by default) the `components_` vectors are divided
-        by n_samples times singular values to ensure uncorrelated outputs
-        with unit component-wise variances.
+    whiten : bool, optional (default False)
+        When True the `components_` vectors are divided by n_samples times
+        singular values to ensure uncorrelated outputs with unit component-wise
+        variances.
 
         Whitening will remove some information from the transformed signal
         (the relative variance scales of the components) but can sometime
         improve the predictive accuracy of the downstream estimators by
-        making there data respect some hard-wired assumptions.
+        making their data respect some hard-wired assumptions.
+
+    svd_solver : string (default 'auto')
+        The algorithm that runs SVD
+        if svd_solver == 'full', run exact SVD calling ARPACK solver via
+        `scipy.linalg.svd` and select the components by postprocessing
+        if svd_solver == 'arpack', run SVD truncated to n_components calling
+        ARPACK solver via `scipy.sparse.linalg.svds`. It requires strictly
+        0 < n_components < X.shape[1]
+        if svd_solver == 'randomized', run randomized SVD by the method of
+        Halko et al.
+        if svd_solver == 'auto', the solver is selected by a default policy
+        based on X.shape and n_components.
+
```

Comment:
These line breaks will not be preserved when building the html docs. [Here's](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/manifold/locally_linear.py#L123) an example of formatting a description list. [Here's](http://scikit-learn.org/stable/modules/generated/sklearn.manifold.LocallyLinearEmbedding.html#sklearn.manifold.LocallyLinearEmbedding) the formatted output.